pkg/promtail/targets/manager.go

```diff
@@ -46,6 +47,22 @@ func NewTargetManagers(
 	}
 	targetManagers = append(targetManagers, fileTargetManager)
 
+	for _, cfg := range scrapeConfigs {
```

Comment:
I think you might want to build the fileTargetManagers the same way? check if the cfg is null and append? then maybe only instantiate one or the other if the list of configs is > 0?

With the journal target manager option I could see people not specifying a filetargetmanager

Reply:
The FileTargetManager reads from the prometheus ServiceDiscoveryConfig which is a non-pointer (and can't be made a pointer due to it being inlined in the yaml). I've added a method to check to see if it's the zero value instead.